builtins/web/timers.cpp

```diff
@@ -12,13 +12,30 @@
 #define S_TO_NS(s) ((s) * 1000000000)
 #define NS_TO_MS(ns) ((ns) / 1000000)
 
+namespace {
+
+class TimersMap {
+public:
+  std::map<int32_t, api::AsyncTask*> timer_ids_ = {};
+  int32_t next_timer_id = 1;
+  void trace(JSTracer *trc) {
+    for (std::map<int32_t, api::AsyncTask*>::iterator i = timer_ids_.begin(); i != timer_ids_.end(); i++) {
+      api::AsyncTask *timer = i->second;
+      timer->trace(trc);
+    }
+  }
+};
+
+}
+
+static TimersMap *TIMERS_MAP;
```

Comment:
Wrapping this in a `PersistentRooted` makes it be traced automatically:

```suggestion
static PersistentRooted<TimersMap*> TIMERS_MAP;
```

Reply:
This doesn't seem to build, giving the error:

```
/Users/gbedford/Projects/StarlingMonkey/deps/cpm_cache/spidermonkey-debug/83add297093a844dd3ba963092e46338080a42b3/spidermonkey-debug/include/js/RootingAPI.h:1399:41: note: in instantiation of template type alias 'RootedTraits' requested here
class PersistentRooted : public detail::RootedTraits<T>::PersistentBase,
                                        ^
/Users/gbedford/Projects/StarlingMonkey/builtins/web/timers.cpp:30:37: note: in instantiation of template class 'JS::PersistentRooted<(anonymous namespace)::TimersMap *>' requested here
static PersistentRooted<TimersMap*> TIMERS_MAP;
```